From f1c03bf225d0099b4adcf18a4ec8ed838c1b4c87 Mon Sep 17 00:00:00 2001
From: elektroschmock <elektroschmock78@googlemail.com>
Date: Fri, 29 Sep 2017 13:42:44 +0200
Subject: [PATCH 06/13] legacy: Allow system-server access to sensor devices

avc: denied { getattr } for pid=797 comm="system-server-i"
path="/dev/sensors"
dev="tmpfs"
ino=8720
scontext=u:r:system_server:s0
tcontext=u:object_r:sensors_device:s0
tclass=chr_file permissive=1

Change-Id: I225b21fc078a68038cd2df8bd5cea3082656a695
---
 legacy-common/system_server.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/legacy-common/system_server.te b/legacy-common/system_server.te
index e0ae2cc..9c9c7f7 100644
--- a/legacy-common/system_server.te
+++ b/legacy-common/system_server.te
@@ -11,6 +11,7 @@ use_per_mgr(system_server)
 allow system_server binder_per_mgr_service:service_manager find;
 
 allow system_server sensors:unix_stream_socket sendto;
+allow system_server sensors_device:chr_file getattr;
 allow system_server sensors_socket:sock_file r_file_perms;
 unix_socket_connect(system_server, sensors, sensors)
 unix_socket_send(system_server, sensors, sensors)
-- 
2.7.4

